server/mqtt.go

```diff
@@ -3584,23 +4015,23 @@ func mqttDeliverMsgCbQos0(sub *subscription, pc *client, _ *Account, subject, re
 	// Check the subscription's QoS. This needs to be protected because
 	// the client may change an existing subscription at any time.
 	sess.mu.Lock()
-	subQos := sub.mqtt.qos
-	isReserved := mqttCheckReserved(sub, subject)
+	subQOS := sub.mqtt.qos
 	sess.mu.Unlock()
 
 	// We have a wildcard subscription and this subject starts with '$' so ignore per Spec [MQTT-4.7.2-1].
-	if isReserved {
+	if mqttIsReserved(sub, subject) {
```

Comment:
No, see comment on that function. It needs the lock. So keep the `isReserved := mqttIsReserved(sub, subject)` inside the lock few lines above.

Reply:
Oops I missed this one, fixed QoS1 only. Thx!!!